libr/bin/format/objc/mach0_classes.c

```diff
@@ -0,0 +1,734 @@
+#include "mach0_classes.h"
+
+#define RO_META               (1<<0)
+
+Mach0_struct_(SMethodList) {
+	ut32 entsize;
+	ut32 count;
+	/* SMethod first;  These structures follow inline */
+};
+
+Mach0_struct_(SMethod) {
+	mach0_ut name;	/* SEL (32/64-bit pointer) */
+	mach0_ut types;	/* const char * (32/64-bit pointer) */
+	mach0_ut imp;		/* IMP (32/64-bit pointer) */
+};
+
+Mach0_struct_(SClass) {
+	mach0_ut isa;			/* SClass* (32/64-bit pointer) */
+	mach0_ut superclass;	/* SClass* (32/64-bit pointer) */
+	mach0_ut cache;		/* Cache (32/64-bit pointer) */
+	mach0_ut vtable;		/* IMP * (32/64-bit pointer) */
+	mach0_ut data;		/* SClassRoT * (32/64-bit pointer) */
+};
+
+Mach0_struct_(SClassRoT) {
+	ut32 flags;
+	ut32 instanceStart;
+	ut32 instanceSize;
+#ifdef R_MACH0_CLASS64
+	ut32 reserved;
+#endif
+	mach0_ut ivarLayout;		/* const uint8_t* (32/64-bit pointer) */
+	mach0_ut name;			/* const char* (32/64-bit pointer) */
+	mach0_ut baseMethods; 	/* const SMEthodList* (32/64-bit pointer) */
+	mach0_ut baseProtocols; 	/* const SProtocolList* (32/64-bit pointer) */
+	mach0_ut ivars; 		/* const SIVarList* (32/64-bit pointer) */
+	mach0_ut weakIvarLayout; 	/* const uint8_t * (32/64-bit pointer) */
+	mach0_ut baseProperties; 	/* const SObjcPropertyList* (32/64-bit pointer) */
+};
+
+Mach0_struct_(SProtocolList) {
+	mach0_ut count;	/* uintptr_t (a 32/64-bit value) */
+	/* SProtocol* list[0];  These pointers follow inline */
+};
+
+Mach0_struct_(SProtocol) {
+	mach0_ut isa;						/* id* (32/64-bit pointer) */
+	mach0_ut name;					/* const char * (32/64-bit pointer) */
+	mach0_ut protocols;				/* SProtocolList*
+									(32/64-bit pointer) */
+	mach0_ut instanceMethods;			/* SMethodList* (32/64-bit pointer) */
+	mach0_ut classMethods;			/* SMethodList* (32/64-bit pointer) */
+	mach0_ut optionalInstanceMethods;	/* SMethodList* (32/64-bit pointer) */
+	mach0_ut optionalClassMethods;	/* SMethodList* (32/64-bit pointer) */
+	mach0_ut instanceProperties;		/* struct SObjcPropertyList*
+									(32/64-bit pointer) */
+};
+
+Mach0_struct_(SIVarList) {
+	ut32 entsize;
+	ut32 count;
+	/* SIVar first;  These structures follow inline */
+};
+
+Mach0_struct_(SIVar) {
+	mach0_ut offset;		/* uintptr_t * (32/64-bit pointer) */
+	mach0_ut name;		/* const char * (32/64-bit pointer) */
+	mach0_ut type;		/* const char * (32/64-bit pointer) */
+	ut32 alignment;
+	ut32 size;
+};
+
+Mach0_struct_(SObjcProperty) {
+	mach0_ut name;		/* const char * (32/64-bit pointer) */
+	mach0_ut attributes;	/* const char * (32/64-bit pointer) */
+};
+
+Mach0_struct_(SObjcPropertyList) {
+	ut32 entsize;
+	ut32 count;
+	/* struct SObjcProperty first;  These structures follow inline */
+};
+
+///////////////////////////////////////////////////////////////////////////////
+static mach0_ut Mach0_class_(get_pointer)(mach0_ut p, ut32 *offset, ut32 *left,
```

Comment:
I mean the declaration of all these static functions. You can just use them without declaring them if they are in the right order. Also, as said in another comment, note that you can avoid the Mach0_class_ macro for static functions